### DISCUSSION TITLE
Ring Doorbell Chimes Not Playing on HomePods

### DISCUSSION THREAD

**Discussion by Cosimoto1** [dgreif/ring#1500] [Q&A]
Recently, my Ring doorbell chime has stopped playing through my HomePods. Haven’t changed anything that I’m aware of and have tried reinstalling the Ring Homebridge plugin. But still cannot get the chimes to play. Any idea what I’m doing wrong? I am attaching my log. 
[homebridge-ring.log.txt](https://github.com/user-attachments/files/17183051/homebridge-ring.log.txt)

**Comment by tsightler** [comment]
Did you follow the steps in the release notes for v13 or any of the troubleshooting steps from [the wiki](https://github.com/dgreif/ring/wiki/Notification-Troubleshooting)?  There are really only two options that it can be, either there are no entries in the homebridge-ring logs when a ding is detected, then the wiki/release notes troubleshooting steps should correct it, or there are log entries but there are no dings in Homekit, which is more difficult since that means the problem is something in Homekit and we can't really help a lot with that.

**Comment by Cosimoto1** [comment]
Thanks for the quick reply. I’ll give the troubleshoot steps in the wiki a shot.

**Comment by Cosimoto1** [comment]
Can you tell me where the QR code to add the Ring HomeBridge plugin is located? I must be an idiot because I cannot find it.

**Reply by tsightler** [reply]
It's in the Homebridge UI, or you can also find it in the logs if you just want to manually enter the number (the QR code is just a faster way to enter the paring code, but you can also just type it manually as it's typically only 8 digits or so).

**Comment by thrangster** [comment]
Hi - I'm trying to add the feature to you describe in the Wiki to subscribe to the token refresh:

  ringApi.onRefreshTokenUpdated.subscribe(
    async ({ newRefreshToken, oldRefreshToken }) => {
      console.log('Refresh Token Updated: ', newRefreshToken)


      // If you are implementing a project that use `ring-client-api`, you should subscribe to onRefreshTokenUpdated and update your config each time it fires an event
      // Here is an example using a .env file for configuration
      if (!oldRefreshToken) {
        return
      }


      const currentConfig = await promisify(readFile)('.env'),
        updatedConfig = currentConfig
          .toString()
          .replace(oldRefreshToken, newRefreshToken)


      await promisify(writeFile)('.env', updatedConfig)
    }
  )

But I do not know exactly there to paste this in my config, as I am getting invalid JSON errors - I am not a programmer so sorry, this is confusing.... this is my Ring Homebridge config (identifying information remove in this post):

{
    "refreshToken": "<<<<<REMOVED>>>>>>"
    "hideDoorbellSwitch": true,
    "hideCameraLight": true,
    "hideCameraSirenSwitch": true,
    "hideInHomeDoorbellSwitch": false,
    "hideAlarmSirenSwitch": true,
    "hideDeviceIds": [
        "<<<<REMOVED>>>>"
    ],
    "_bridge": {
        "username": "<<<REMOVE>>>",
        "port": 34627
    },
    "platform": "Ring"
}

Where does it go in relation to the brackets in my existing config as I'm getting JSON error when pasting at the bottom. Also do I have to personalize anything in the same code you provided?

The issue is the doorbell fails to ring on my HomePods after a short while, so presuming this is the issue.


Thanks kindly
